inference-engine/src/inference_engine/include/ie/ie_core.hpp

```diff
@@ -107,7 +107,7 @@ class INFERENCE_ENGINE_API_CLASS(Core) {
      * @return An executable network reference
      */
     ExecutableNetwork LoadNetwork(const CNNNetwork& network,
-                                  const std::string& deviceName,
+                                  const std::string& deviceName = "AUTO",
```

Comment:
from whta I see, only the old API is changed.
the OV 2.0 API should be covered as well: inference-engine/src/inference_engine/include/openvino/runtime/core.hpp

Reply:
@songbell please modify compile_model for OV API 2.0 as Maxim mentioned.

Reply:
is it OK that if AUTO plugin is not in deployment package, it will just throw an exception that AUTO is not registered in Core?

Reply:
Good idea, Ilya. We will double check.

Reply:
@ilya-lavrenov, as below link shows, it will do throw an exception when AUTO is not registered in Core. 
https://github.com/songbell/openvino/blob/b51c930a553e01a64ac6061c07f21968c32acbfa/inference-engine/src/inference_engine/src/ie_core.cpp#L544

Reply:
The test case for invalid device is also covered by below PR, including passing empty("") device to LoadNetwork.
https://github.com/openvinotoolkit/openvino/pull/8786

Reply:
@ilya-lavrenov does this resolves your concern?

Reply:
maybe the concern is that user haven't specified the name, but will get an error on some "AUTO" device is missing? so that it is better to add a more descriptive error message like "No device is provided, so AUTO device is used by default, whihc failed loading. Should you rebuild OpenVINO with the AUTO enabled or re-install accordingly? "